R-package/R/utils.R

```diff
@@ -74,6 +74,18 @@ check.booster.params <- function(params, ...) {
         params[['monotone_constraints']] = vec2str
   }
   
+  # interaction constraints parser
+  if (!is.null(params[['interaction_constraints']])){
```

Comment:
To be consistent with monotone_constraints, let's also allow interaction_constraints to be passed as a character string.

Reply:
done